trustfall_core/src/schema/adapter/mod.rs

```diff
@@ -214,15 +215,62 @@ impl<'a> crate::interpreter::Adapter<'a> for SchemaAdapter<'a> {
         &self,
         edge_name: &Arc<str>,
         _parameters: &EdgeParameters,
-        _resolve_info: &ResolveInfo,
+        resolve_info: &ResolveInfo,
     ) -> VertexIterator<'a, Self::Vertex> {
+        let candidate_value = resolve_info.statically_required_property("name").clone();
         match edge_name.as_ref() {
             "VertexType" => {
                 let root_query_type = self.schema.query_type_name();
-                Box::new(self.schema.vertex_types.values().filter_map(move |v| {
-                    (v.name.node != root_query_type)
-                        .then(|| SchemaVertex::VertexType(VertexType::new(v)))
-                }))
+
+                if let Some(crate::interpreter::CandidateValue::Single(FieldValue::String(
```

Comment:
Can you add a `use` for `CandidateValue` so that these lines don't end up being so long?

Reply:
Done!